yt_dlp/extractor/qqmusic.py

```diff
@@ -1,245 +1,314 @@
-            json_obj_all_songs = self._parse_json(json_text, singmid)
-
-        for item in json_obj_all_songs['data']['list']:
-            if item['musicData'].get('songmid') is not None:
-                songmid = item['musicData']['songmid']
-                entries.append(self.url_result(
-                    rf'https://y.qq.com/n/yqq/song/{songmid}.html', 'QQMusic', songmid))
-
-        return entries
-
-
-class QQMusicSingerIE(QQPlaylistBaseIE):
+class QQMusicSingerIE(QQMusicBaseIE):
     IE_NAME = 'qqmusic:singer'
     IE_DESC = 'QQ音乐 - 歌手'
-    _VALID_URL = r'https?://y\.qq\.com/n/yqq/singer/(?P<id>[0-9A-Za-z]+)\.html'
-    _TEST = {
-        'url': 'https://y.qq.com/n/yqq/singer/001BLpXF2DyJe2.html',
+    _VALID_URL = r'https?://y\.qq\.com/n/ryqq/singer/(?P<id>[0-9A-Za-z]+)'
+    _TESTS = [{
+        'url': 'https://y.qq.com/n/ryqq/singer/001BLpXF2DyJe2',
         'info_dict': {
             'id': '001BLpXF2DyJe2',
             'title': '林俊杰',
-            'description': 'md5:870ec08f7d8547c29c93010899103751',
+            'description': 'md5:10624ce73b06fa400bc846f59b0305fa',
+            'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
         },
-        'playlist_mincount': 12,
-    }
+        'playlist_mincount': 100,
+    }, {
+        'url': 'https://y.qq.com/n/ryqq/singer/000Q00f213YzNV',
+        'info_dict': {
+            'id': '000Q00f213YzNV',
+            'title': '桃几OvO',
+            'description': '小破站小唱见~希望大家喜欢听我唱歌~！',
+            'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
+        },
+        'playlist_count': 12,
+        'playlist': [{
+            'info_dict': {
+                'id': '0016cvsy02mmCl',
+                'ext': 'mp3',
+                'title': '群青',
+                'album': '桃几2021年翻唱集',
+                'release_date': '20210913',
+                'duration': 248,
+                'creators': ['桃几OvO'],
+                'genres': ['Pop'],
+                'description': 'md5:4296005a04edcb5cdbe0889d5055a7ae',
+                'size': 3970822,
+                'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
+            },
+        }],
+    }]
+
+    def _entries(self, mid, init_data):
+        page_size = 50
+        max_num = traverse_obj(init_data, ('singerDetail', 'songTotalNum'))
+        for page in range(max_num // page_size + 1):
+            data = self.make_fcu_req({'req_1': {
+                'module': 'music.web_singer_info_svr',
+                'method': 'get_singer_detail_info',
+                'param': {
+                    'sort': 5,
+                    'singermid': mid,
+                    'sin': page * page_size,
+                    'num': page_size,
+                }}}, mid, note=f'Downloading page {page}')
+            yield from traverse_obj(data, ('req_1', 'data', 'songlist', ..., {lambda x: self.url_result(
+                f'https://y.qq.com/n/ryqq/songDetail/{x["mid"]}', QQMusicIE, x['mid'], x.get('title'))}))
```

Comment:
entries could be an `OnDemandPagedList`

Reply:
https://github.com/yt-dlp/yt-dlp/pull/9768/commits/8944cae2ff2860513b77abe9ec49f09995d49233